.github/workflows/ping-maintainers.yml

```
@@ -0,0 +1,72 @@
+---
+name: Ping Maintainers
+
+on:  # yamllint disable-line rule:truthy
+  pull_request:
+    types: [opened, ready_for_review, synchronize, reopened, edited]
+    paths:
+      - 'modules/'
+
+jobs:
+  ping-maintainers:
+    runs-on: ubuntu-24.04
+    if: github.repository_owner == 'danth'
+    steps:
+      - uses: actions/checkout@v4.2.2
+
+      - uses: DeterminateSystems/nix-installer-action@v16
+
+      - uses: cachix/cachix-action@v16
+        with:
+          name: stylix
+          authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
+        continue-on-error: true
+
+      - run: nix build .#get-maintainers
+
+      - id: changed-files
+        uses: jitterbit/get-changed-files@v1
+        with:
+          format: 'json'
+
+      - id: get-maintainers
+        env:
+          PR_AUTHOR: ${{ github.event.pull_request.user.login }}
+          REVIEWERS_API_URL: https://api.github.com/repos/${{ github.repository }}/pulls/${{ github.event.pull_request.number }}/requested_reviewers  # yamllint disable-line rule:line-length
+        run: |
+          MAX_MAINTAINERS=5
+
+          existing_reviewers="$(
+            curl --silent "$REVIEWERS_API_URL" |
+              jq --raw-output [.users[].login]
+          )"
+
+          to_ping=()
+
+          jq --raw-output '.[]' <<<"$CHANGED_FILES" | while read -r file; do
+            jq --arg file "$file" --raw-output '.[$file][]' result/gh_ids.json |
+              while read -r gh_id;
+            do
+              # Do not request the original author or already requested
+              # reviewers.
+              if [[
+                "$gh_id" == "$PR_AUTHOR" || \
+                jq "index(\"$gh_id\") != null" <<< "$existing_reviewers"
```

Nitpick:

```suggestion
                jq "index(\"$gh_id\") != null" <<<"$existing_reviewers"
```

I did this incorrectly in https://github.com/danth/stylix/pull/1053#discussion_r2075576415.